.githooks/pre-push.bash

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Check if we actually have commits to push
+commits=$(git log @{u}..)
+if [ -z "$commits" ]; then
+    exit 0
+fi
+
+if ! command -v golangci-lint &> /dev/null; then
+    echo "installing golangci-lint..."
+    go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
+fi
+
+make lint
+
+make vet
+
+make test-race
+
+exit 0
```

Comment:
`exit 0` should be implicit, just curious why it is needed?

Reply:
probably not, I like it like that :)
will check without it